dyne/ukf.py

```diff
@@ -5,6 +5,24 @@
 from ._common import check_input_arrays, check_measurements
 
 
+def _unscented_transform(x, P, Q, alpha):
+    n_states = len(x)
+    n_noises = len(Q)
+    l, V = linalg.eigh(P)
+    # Due to numerical issues computed eigenvalues could be negative,
+    # here we set them to zero
+    l[l < 0] = 0
+    Sigma = alpha * np.sqrt(n_states + n_noises) * linalg.block_diag(
+        np.diag(l**0.5) @ V.T, linalg.cholesky(Q))
+    X_probe = []
+    for sigma in Sigma:
+        for sign in [-1, 1]:
+            X_probe.append(list(x + sign * sigma[:n_states])
```

Comment:
Use `np.hstack` for such kind of operations.

Reply:
good comment